extensions/amp-story/1.0/amp-story.js

```diff
@@ -1470,6 +1550,72 @@ export class AmpStory extends AMP.BaseElement {
     });
   }
 
+  /** @private */
+  onHideSubscriptionsDialog_() {
+    this.paywallTimeout_ && clearTimeout(this.paywallTimeout_);
+    this.paywallTimeout_ = null;
+    if (
+      this.pageAfterGranted_ &&
+      this.storeService_.get(StateProperty.SUBSCRIPTIONS_STATE) ===
+        SubscriptionsState.GRANTED
+    ) {
+      this.switchTo_(this.pageAfterGranted_, NavigationDirection.NEXT);
+    }
+    this.pageAfterGranted_ = null;
+  }
+
+  /**
+   * @return {boolean}
+   * @private
+   */
+  isPaywallStory_() {
```

Comment:
You could replace this with a new state `SubscriptionsState.NO_SUBSCRIPTIONS` or something. You'd save the DOM query + simplify the `if` tests in `switchTo()`